travel-event-host-app/src/app/api/user/[id]/route.ts

```diff
@@ -0,0 +1,22 @@
+import { NextResponse } from 'next/server';
+import { connectMongoDB } from '@/lib/mongodb';
+import mongoose from 'mongoose';
+import User from '@/models/user';
+
+export async function GET(req: Request, { params }: any) {
+  //to get query use this, here you get query1 for example
+  const { searchParams } = new URL(req.url);
+  const query1 = searchParams.get('query1');
+  await connectMongoDB();
+
+  const id = params.id;
+  const isValidObjectId = mongoose.Types.ObjectId.isValid(id);
+  if (!isValidObjectId) {
+    return NextResponse.json({ message: 'Invalid ObjectId format' }, { status: 400 });
+  }
+
+  const userFound = await User.findById(id);
+  console.log(typeof userFound);
+  if (userFound) return NextResponse.json({ userFound }, { status: 201 });
```

Comment:
Since we're returning the entire user object we retrieve from the DB, it's also going to return sensitive info like the hashed password. I think we need to return an object that does not have properties that are sensitive. You might want to create an interface that has the sensitive info filtered out of the response

Reply:
what info we want to filter?

Reply:
`email`, `password`, `admin`